versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/BaseCommitHelper.java

```diff
@@ -230,6 +218,142 @@ static <R, I> R committingOperation(
     }
   }
 
+  void validateNamespacesHaveNoChildren(
+      Set<ContentKey> deletedNamespaces, StoreIndex<CommitOp> headIndex)
+      throws ReferenceConflictException {
+    if (!persist.config().validateNamespaces()) {
+      return;
+    }
+
+    for (ContentKey namespaceKey : deletedNamespaces) {
+      StoreKey storeKey = keyToStoreKey(namespaceKey);
+
+      for (Iterator<StoreIndexElement<CommitOp>> iter = headIndex.iterator(storeKey, null, false);
+          iter.hasNext(); ) {
+        StoreIndexElement<CommitOp> el = iter.next();
+        if (!el.content().action().exists()) {
+          // element does not exist - ignore
+          continue;
+        }
+        ContentKey elContentKey = storeKeyToKey(el.key());
+        if (elContentKey == null) {
+          // not a "content object" - ignore
+          continue;
+        }
+        if (elContentKey.equals(namespaceKey)) {
+          // this is our namespace - ignore
+          continue;
+        }
+
+        int elLen = elContentKey.getElementCount();
+        int nsLen = namespaceKey.getElementCount();
+
+        // check if element is in the current namespace, fail it is true - this means,
+        // there is a live content-key in the current namespace - must not delete the namespace
+        if (elLen >= nsLen
+            && elContentKey.getElements().subList(0, nsLen).equals(namespaceKey.getElements())) {
+          throw new ReferenceConflictException(
+              format(
+                  "The namespace '%s' would be deleted, but cannot, because it has children.",
+                  namespaceKey));
+        }
+      }
+    }
+  }
+
+  void validateNamespacesExistForContentKeys(
+      Map<ContentKey, Content> newContent, StoreIndex<CommitOp> headIndex)
+      throws ReferenceConflictException {
+    Set<ContentKey> namespaceKeys =
+        newContent.keySet().stream()
+            .filter(k -> k.getElementCount() > 1)
+            .map(ContentKey::getParent)
+            .collect(Collectors.toSet());
+    if (!persist.config().validateNamespaces()) {
+      return;
+    }
+
+    for (ContentKey key : namespaceKeys) {
+      Content namespaceAddedInThisCommit = newContent.get(key);
+      if (namespaceAddedInThisCommit instanceof Namespace) {
+        // Namespace for the current new-content-key has been added via the currently validated
+        // commit. Nothing to do for `Namespace`s here.
+        return;
```

Comment:
Why `return` and not `continue`? We could have other namespaces referenced, but not created in this commit.

Reply:
duh!

Reply:
fixed